rest_framework/serializers.py

```diff
@@ -1131,9 +1131,14 @@ def build_relational_field(self, field_name, relation_info):
         field_kwargs = get_relation_kwargs(field_name, relation_info)
 
         to_field = field_kwargs.pop('to_field', None)
-        if to_field and not relation_info.related_model._meta.get_field(to_field).primary_key:
-            field_kwargs['slug_field'] = to_field
-            field_class = self.serializer_related_to_field
+        if relation_info.reverse:
+            if to_field and not relation_info.related_model_field.related_fields[0][1].primary_key:
+                field_kwargs['slug_field'] = to_field
+                field_class = self.serializer_related_to_field
```

Comment:
Yeah, it looks like we need to add a test case where the ```to_field``` argument on the foreign key relationship is set to a non-pk field in order to cover this branch.